ext/rugged/rugged_remote.c

```diff
@@ -620,6 +620,49 @@ static VALUE rb_git_remote_push(int argc, VALUE *argv, VALUE self)
 	return payload.result;
 }
 
+/*
+ *  call-seq:
+ *    remote.prune(options = {}) -> nil
+ *
+ *  Prune tracking refs that are no longer present on remote
+ *
+ *  Returns nil
+ *
+ *  The following options can be passed in the +options+ Hash:
+ *
+ *  :credentials ::
+ *    The credentials to use for the fetch operation. Can be either an instance of one
+ *    of the Rugged::Credentials types, or a proc returning one of the former.
+ *    The proc will be called with the +url+, the +username+ from the url (if applicable) and
+ *    a list of applicable credential types.
+ *
+ *  Example:
+ *
+ *    remote = Rugged::Remote.lookup(@repo, 'origin')
+ *    remote.prune()
+ *
+ */
+static VALUE rb_git_remote_prune(int argc, VALUE *argv, VALUE self)
+{
+	VALUE rb_options;
+	git_remote *remote;
+	git_fetch_options opts = GIT_FETCH_OPTIONS_INIT;
+	struct rugged_remote_cb_payload payload = { Qnil, Qnil, Qnil, Qnil, Qnil, rb_hash_new(), 0 };
```

Comment:
This should be `struct rugged_remote_cb_payload payload = { Qnil, Qnil, Qnil, Qnil, Qnil, Qnil, 0 };`. I don't think the hash object created here is used anywhere.